Fixed dialog windows tiling like main windows on tiling window managers. (#464)
TagStudioDev/TagStudio: CONTRIBUTING.md

> [!TIP]
> On Linux and macOS, you can launch the `tagstudio.sh` script to skip the following process, minus the `requirements-dev.txt` installation step. _Using the script is fine if you just want to launch the program from source._

1. Make sure you're using the correct Python version:
   `python3 --version` (As described above)

>    -   If the output matches `Python 3.12.x` where the x is any number representing a patch update (Does not matter), then continue to step 2 and ignore the info below about pyenv, you won't need to install it either.
>    -   Otherwise, you probably want to use pyenv so you don't have to downgrade or upgrade your Python installation before continuing to step 2:

Reviewer: Recommend making the working here more succinct and give the user a choice of how they'd like to install the required Python version:
```suggestion
   -   If the output matches `Python 3.12.x` (where the x is any number) then you're using the correct Python version and can skip to Step 2. Otherwise you can install the required Python version from the [Python](https://www.python.org/downloads/) website or use a tool such as [pyenv](https://github.com/pyenv/pyenv/) to install an additional version of Python without changing the existing version on your system. 
```

>       1. Follow the install instructions for your system on [the pyenv Github repository](https://github.com/pyenv/pyenv/).

Reviewer: Recommend linking directly to the install instructions
```suggestion
      1. Follow pyenv's [install instructions](https://github.com/pyenv/pyenv/?tab=readme-ov-file#installation) for your system.
```

>       2. Install the appropriate Python version with pyenv by running `pyenv install 3.12` (This will **not** mess with your existing Python install in any way).

Reviewer: ```suggestion
      2. Install the appropriate Python version with pyenv by running `pyenv install 3.12` (This will **not** mess with your existing Python installation).
```

>       3. Go into the root of this repository with a terminal, and run `pyenv local 3.12` to tell pyenv that Python 3.12 should be used in this folder (This will **not** mess with your existing Python install either, it only applies to this folder).

Reviewer: ```suggestion
      3. Navigate to the repository root folder in your terminal and run `pyenv local 3.12`.
```

>       - You can also do `pyenv shell 3.12` or `pyenv global 3.12` instead to set the Python version for the current terminal session or the entire system respectively, but `local` is recommended.

Reviewer: ```suggestion
      - You could alternatively use `pyenv shell 3.12` or `pyenv global 3.12` to set the Python version for the current terminal session or the entire system respectively, however using `local` is recommended.
```

2. In the root repository directory, create a python virtual environment:  
   `python3 -m venv .venv`
3. Activate your environment:

-   Windows w/Powershell: `.venv\Scripts\Activate.ps1`
-   Windows w/Command Prompt: `.venv\Scripts\activate.bat`
-   Linux/macOS: `source .venv/bin/activate`
       If you use an alternative shell like fish or csh, you may wish to switch to the default shell (Usually bash) for this project, or alternatively, you can see the other activation scripts left by Python such as:
       - Fish: `source .venv/bin/activate.fish`

>        - CSH: `source .venv/bin/activate.csh`

Reviewer: The inclusion of the other activation scripts is nice, but why the recommendation for the default shell/bash? Personally I've never ran into an issue with using fish for this project

Author: Actually, I did run into syntax issues with it. Shell scripts are supposed to specify which shell implementation they should run with, but on my machine, the regular activation script doesn't for whatever reason, so it defaults to the shell you're running with. Perhaps it's just different across distros for some reason, but on my end it doesn't specify, it simply has a comment at the top saying it "must be used directly with bash."

![image](https://github.com/user-attachments/assets/0a701e4b-9305-4d02-9069-7d6e8ddaff6b)
![image](https://github.com/user-attachments/assets/92d8d6d4-1415-4f0d-a71f-b31dd08def97)

As for the recommendation, I guess I just didn't word it right. I meant to say switching to bash is an option.

4. Install the required packages:

-   `pip install -r requirements.txt`
-   If developing (includes Ruff and Mypy): `pip install -r requirements-dev.txt`

_Learn more about setting up a virtual environment [here](https://docs.python.org/3/tutorial/venv.html)._

### Manually Launching (Outside of an IDE)

> #### If you encounter errors about the Python version, or seemingly vague script errors, [pyenv](https://github.com/pyenv/pyenv/) may solve your issue, see step 1 of [Creating a Python Virtual Environment](#creating-a-python-virtual-environment)

Reviewer: ```suggestion
If you encounter errors about the Python version, [pyenv](https://github.com/pyenv/pyenv/) may solve your issue. See Step 1 of [Creating a Python Virtual Environment](#creating-a-python-virtual-environment).
```
